File: 프로그래머스/level0/안전지대.py
def solution(board):
    new_list = [[i, j] for i in range(len(board)) for j in range(len(board[i])) if board[i][j] == 1]
    answer = len(new_list)
    for n in new_list:
        for i in [-1, 0, 1]:
            for j in [-1, 0, 1]:
                if 0 <= n[0] + i < len(board) and 0 <= n[1] + j < len(board):
                    if board[n[0] + i][n[1] + j] == 0:
                        board[n[0] + i][n[1] + j] = 1
                        answer += 1
    return len(board) ** 2 - answer

File: 프로그래머스/level0/안전지대.java
package 프로그래머스.level0;

public class 안전지대 {
    public int solution(int[][] board) {
        int answer = 0;
        int[][] newBoard = new int[board.length + 2][board.length + 2];
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if(board[i][j] == 1){
                    for (int a = i; a <= i + 2; a++) {
                        for (int b = j; b <= j + 2; b++) {
                            if (newBoard[a][b] != 1) {
                                newBoard[a][b] = 2;
                            }
                        }
                    }
                }
            }
        }

        for (int i = 1; i < newBoard.length - 1; i++) {
            for (int j = 1; j < newBoard.length - 1; j++) {
                if(newBoard[i][j] == 0)
                    answer++;
            }
        }
        return answer;
    }
}
